src/Microsoft.AspNetCore.Server.IIS/Core/IO/AsyncIOOperation.cs

```diff
@@ -103,7 +104,8 @@ public AsyncContinuation Complete(int hr, int bytes)
                 _result = bytes;
                 if (hr != NativeMethods.HR_OK)
                 {
-                    _exception = new IOException("Native IO operation failed", Marshal.GetExceptionForHR(hr));
+                    // Treat all errors as the client disconnect
+                    _exception = new ConnectionResetException("The client disconnected", Marshal.GetExceptionForHR(hr));
```

Comment:
nit: The client has disconnected.

Reply:
In my defence - I copied it from System.Web.